### ISSUE TITLE
Error page of kubernetes ingress default route not showing

### ISSUE THREAD

**Issue by geekgonecrazy** [containous/traefik#3810]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
```yml
apiVersion: v1
kind: Service
metadata:
  name: default-route
  namespace: my-namespace
  annotations:
    traefik.backend.circuitbreaker: "NetworkErrorRatio() > 0.5"
spec:
  ports:
  - name: http
    targetPort: 5000
    port: 5000
  selector:
    app: default-route

---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: default-route
  namespace: my-namespace
  annotations:
    kubernetes.io/ingress.class: "traefik"
    traefik.ingress.kubernetes.io/error-pages: |-
        foo:
          status:
          - "400-600"
          backend: default-route
          query: "/{status}.html"
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: default-route
          servicePort: 5000
```
I've also tried setting with priority no luck.

### What did you expect to see?
If I have an ingress route and service  defined.. and I  remove all pods.  I would expect to see my default route error page instead of `service unavailable`

### What did you see instead?
`service unavailable`

### Output of `traefik version`: (_What version of Traefik are you using?_)
docker image: `traefik:v1.6.5`

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Running as kubernetes ingress provider

```toml
defaultEntryPoints = ["http","https"]
    [entryPoints]
      [entryPoints.http]
      address = ":80"
        [entryPoints.http.redirect]
          entryPoint = "https"
      [entryPoints.https]
      address = ":443"
        [entryPoints.https.tls]
          [[entryPoints.https.tls.certificates]]
          CertFile = "/ssl/wildcard.crt"
          KeyFile = "/ssl/wildcard.key"
      [entryPoints.api]
      address = ":8080"
      [entryPoints.ping]
      address = ":8082"

    [kubernetes]

    [api]
    entryPoint = "api"
    dashboard = true

      [api.statistics]

    [metrics]
      [metrics.prometheus]
        entryPoint = "api"

    [ping]
    entryPoint = "ping"
```

**Comment by kachkaev** [comment]
@dduportal since you are planning to look at this (which is great), how hard would it be to also implement https://github.com/containous/traefik/issues/3420 while your brain is in the realm of error pages? 🙂

**Comment by geekgonecrazy** [comment]
#3678 looks like trying to achieve same thing. 

Basically my setup is a default-route service.   That service responds to any request to *.mydomain.com that doesn't have an ingress route defined for it.

With the addition of error pages I intended to make the default-route service handle the cases where an ingress route has been added but the service wasn't ready yet: `service unavailable` or `bad gateway`.  

So my thought is.. can't reach the service... I should be able to make it hit the default-route and show an error page.  That way at least I can communicate something friendly.

Otherwise I would have to add the exact same error page annotation to 2k some different ingress routes.

Maybe i'm trying to do something that isn't supported?  If so I guess I need to convert this to a feature request :)

**Comment by dtomcej** [comment]
@geekgonecrazy You have a couple things rolled into one with this issue:

1. You are trying to create a global default backend ingress. This is already supported in 1.7: (https://docs.traefik.io/v1.7/configuration/backends/kubernetes/#global-default-backend-ingresses). Trying to implement it in 1.6 will most likely not work (hence why the code changes were required).
2. The error page configuration tells traefik what to render when your ingress has issues. Right now you have your default backend using your default backend to render its own error pages. I don't think that is what you intend to do.
3. You can achieve what you want to do, but yes, you require your ingresses to each specify their own error page configuration. You can modify templates to reduce your coding time, but there is no global "override" for per-ingress configuration.

**Comment by geekgonecrazy** [comment]
Ah... Then yes I guess what i'm attempting to do would be considered global error pages.

By templates do you mean traefik's? From the documentation: https://docs.traefik.io/configuration/commons/#override-default-configuration-template looks like I can only do so if I know ahead of time the frontends?  Though seems a bit vague beyond that like more _**might**_ be possible.

Being able to define a template that would be applied to all frontends would definitely be a cleaner solution

**Comment by geekgonecrazy** [comment]
ah... https://github.com/containous/traefik/blob/master/templates/kubernetes.tmpl looks like this basically will allow me to overwrite this file right?  :thinking:

**Comment by nmengin** [comment]
Hello @geekgonecrazy ,

Many thanks for your feedback.
Indeed, there was a bug with the error pages management.
This bug has been fixed in 1.7 but, your workaround is the best solution to solved it in older versions.

The question seems to be answred.

I close the issue.

**Comment by kachkaev** [comment]
@nmengin I'm not sure if error pages work at all in 1.7, see https://github.com/containous/traefik/issues/3643#issuecomment-411922387. Or has this got fixed already? The last version I tested was 1.7.0-rc.3 and I can't recall any related PRs since then.

**Comment by ldez** [comment]
@kachkaev the bug in v1.7 is not fixed, I'm working on it.
